nixos/modules/services/computing/slurm/slurm.nix

```diff
@@ -369,41 +381,51 @@ in
       wants = [ "network-online.target" ];
 
       serviceConfig = {
-        Type = "forking";
-        KillMode = "process";
-        ExecStart = "${wrappedSlurm}/bin/slurmd";
-        PIDFile = "/run/slurmd.pid";
+        Delegate="Yes";
         ExecReload = "${pkgs.coreutils}/bin/kill -HUP $MAINPID";
+        # The 23.02.06.01 release assumes slurm*d run as Type = "simple",
+        # with the `-D` flag ensuring the daemons stay in foreground.
+        # Already in 23.11.0.0-rc1 the template .service files
+        # suggest to use Type = "notify", and introduce the `--systemd`
+        # flag instead of `-D`. We'll want to update this line with the release:
+        ExecStart = "${wrappedSlurm}/bin/slurmd -D";
+        KillMode = "process";
         LimitMEMLOCK = "infinity";
-        Delegate="Yes";
+        LimitNOFILE = 131072;
+        LimitSTACK = "infinity";
+        RuntimeDirectoryMode = "0755";
+        RuntimeDirectory = "slurm";
+        Type = "simple";
       };
     };
 
-    systemd.tmpfiles.rules = mkIf cfg.client.enable [
-      "d /var/spool/slurmd 755 root root -"
```

Comment:
In the past slurmd had troubles starting if the spooler directory did not exist. Have checked this works reliably?

Reply:
The nixos tests pass, but maybe I revert this.

The home directories are created by the activation script at this point, I think. After consulting with the "nixos contributions" chat, I see that the proper solution would be to use `StateDirectory`: https://www.freedesktop.org/software/systemd/man/latest/systemd.exec.html#Implicit%20Dependencies. This option, however, only accepts relative paths, and creates the respective directories in `/var/lib`.

I think we should do the transition, but I do not want to bother with the deprecation at this point. I'll just recover the tmpfiles rule

Reply:
Dropped the diff